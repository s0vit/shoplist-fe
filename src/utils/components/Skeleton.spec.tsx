import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import SkeletonGroup from 'src/utils/components/Skeleton.tsx';

describe('Skeleton component', () => {
  it('renders correctly', () => {
    const { container } = render(<SkeletonGroup />);
    const skeletons = container.querySelectorAll('[data-testid^="skeletonExpenses"]');
    expect(skeletons.length).toBe(3);
  });
  it('renders correct number of skeletons', () => {
    const { container } = render(<SkeletonGroup count={5} />);
    const skeletons = container.querySelectorAll('[data-testid^="skeletonExpenses"]');
    expect(skeletons.length).toBe(5);
  });
  it('matches snapshot', () => {
    const rendered = render(<SkeletonGroup />);
    expect(rendered).toMatchSnapshot();
  });
});
