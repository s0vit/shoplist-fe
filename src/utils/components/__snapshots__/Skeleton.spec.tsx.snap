// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-frmfij cisIhe"
    data-testid="skeletonStack"
    direction="row"
  >
    <div
      class="sc-VILhF hItdIP"
      data-testid="skeletonExpenses-0"
      variant="rectangular"
    />
    <div
      class="sc-VILhF hItdIP"
      data-testid="skeletonExpenses-1"
      variant="rectangular"
    />
    <div
      class="sc-VILhF hItdIP"
      data-testid="skeletonExpenses-2"
      variant="rectangular"
    />
  </div>
</DocumentFragment>
`;
